src/ORM/EagerLoader.php

```diff
@@ -668,6 +669,94 @@ public function loadExternal(SelectQuery $query, array $results): array
+    {
+        $table = $query->getRepository();
+        $external = $this->externalAssociations($table);
+        if (!$external) {
+            return $results;
+        }
+
+        $assocData = [];
+
+        foreach ($external as $meta) {
+            // $contain = $meta->associations();
+            $instance = $meta->instance();
+            $config = $meta->getConfig();
+            $alias = $instance->getSource()->getAlias();
+            $path = $meta->aliasPath();
+
+            if ($instance->requiresKeys($config)) {
+                $source = $instance->getSource();
+                $keys = $instance->type() === Association::MANY_TO_ONE ?
+                    (array)$instance->getForeignKey() :
+                    (array)$instance->getBindingKey();
+
+                $alias = $source->getAlias();
+                $pkFields = [];
+                /** @var string $key */
+                foreach ($keys as $key) {
+                    $pkFields[] = key($query->aliasField($key, $alias));
+                }
+                $assocData[$path] = [
+                    'alias' => $alias,
+                    'pkFields' => $pkFields,
+                ];
+            }
+        }
+
+        $results = (new Collection($results))
+            ->map(function ($row) use ($query, $external, $assocData) {
+                foreach ($external as $meta) {
+                    $path = $meta->aliasPath();
+                    $keys = null;
+                    if (count($assocData[$path]['pkFields']) === 1) {
+                        // Missed joins will have null in the results.
+                        if (array_key_exists($assocData[$path]['pkFields'][0], $row)) {
+                            // Assign empty array to avoid not found association when optional.
+                            if (!isset($row[$assocData[$path]['pkFields'][0]])) {
+                                $keys = [];
+                            } else {
+                                $value = $row[$assocData[$path]['pkFields'][0]];
+                                $keys = [$value => $value];
+                            }
+                        }
+                    } else {
+                        // Handle composite keys.
+                        $collected = [];
+                        foreach ($assocData[$path]['pkFields'] as $key) {
+                            $collected[] = $row[$key];
+                        }
+                        $keys[implode(';', $collected)] = $collected;
+                    }
+
+                    $callback = $meta->instance()->eagerLoader(
+                        $meta->getConfig() + [
+                            'query' => $query,
+                            'contain' => $meta->associations(),
+                            'keys' => $keys,
+                            'nestKey' => $meta->aliasPath(),
+                        ]
+                    );
+
+                    $row = $callback($row);
```

Comment:
Will this call eager loaders once per row?

Reply:
Yes that's the idea, for unbuffered queries all records for the primary query aren't prefetched, hence the eagerloader is run once per row.

That said currently for MySQL making new queries while the original/parent unbuffered query is active is causing `Cannot execute queries while other unbuffered queries are active`, so need to figure out how to deal with that.